internal/alertmanager/currentalert/process.go

```diff
@@ -238,8 +238,56 @@ func addPvc(log logr.Logger, alertLabels model.LabelSet, alertAnnotations model.
 	return nil
 }
 
-func resizePvc(labels model.LabelSet, annotiations model.LabelSet, client client.Client) error {
-	//TODO
+func resizePvc(log logr.Logger, labels model.LabelSet, annotiations model.LabelSet, client client.Client) error {
+	cr, err := k8sutil.GetCr(string(labels["kafka_cr"]), string(labels["namespace"]), client)
+	if err != nil {
+		return err
+	}
+
+	pvc, err := getPvc(string(labels["persistentvolumeclaim"]), string(labels["namespace"]), client)
+	if err != nil {
+		return err
+	}
+
+	for i, broker := range cr.Spec.Brokers {
+		if strconv.Itoa(int(broker.Id)) == pvc.Labels["brokerId"] {
+			var storageConfigs []v1beta1.StorageConfig
+
+			if broker.BrokerConfig == nil {
+				storageConfigs = cr.Spec.BrokerConfigGroups[cr.Spec.Brokers[i].BrokerConfigGroup].StorageConfigs
+			} else {
+				storageConfigs = broker.BrokerConfig.StorageConfigs
+			}
+
+			for i, c := range storageConfigs {
+				if c.MountPath == pvc.Annotations["mountPath"] {
+					size := *c.PvcSpec.Resources.Requests.Storage()
+
+					size.Add(resource.MustParse(string(annotiations["diskSize"])))
+
+					c.PvcSpec.Resources.Requests = corev1.ResourceList{
+						"storage": size,
+					}
+
+					if broker.BrokerConfig == nil {
+						broker.BrokerConfig = &v1beta1.BrokerConfig{
+							StorageConfigs: []v1beta1.StorageConfig{c},
+						}
+					} else {
+						broker.BrokerConfig.StorageConfigs[i] = c
+					}
+				}
+			}
+			cr.Spec.Brokers[i] = broker
+		}
+	}
+
+	err = k8sutil.UpdateCrWithRollingUpgrade(0, cr, client)
```

Comment:
Is this needed so the kafka pods are bounced to pick up the new resized PV? Is that's the case we may want to make this configurable as newer versions of CSI plugins support hot disk resize

Reply:
Also in my environment Insaw that when resizing the pvc (albeit manually manipulating the kafka cr spec and triggering a rolling restart) it may take some time until the underlying disk is resized. We may need to poll the status of the pv and only when fully resized trigger a rolling here

Reply:
The main reason why I'm updating the KafkaCR is in the next reconciliation loop it will attempt to lower the PVC value which isn't supported. I wanted to originally just change the value in the PVC and in the kafka CR without triggering a restart but the update CR method is private and I figured not every cluster supports dynamic resizing. 

In my setup the restart isn't necessary either; I would assume pretty much any CSI device wouldn't need to restart. I figured there was little harm in restarting, in the current proposal im using the reconciliation loop to increase the size of the PVC which I think is safer.

Reply:
I would make the `updateCR` public or create an another wrapper around it instead of using `k8sutil.UpdateCrWithRollingUpgrade. It is meant to update the errorCount value on the CR. Setting that value to zero will cause problems. That should not be changed from its value, or at least not during a resize alert.

We are aware that rolling upgrading a cluster after a pvc resize might not be handled correctly. I suggest the following. I create an Issue something like `Investigate whether rolling upgrade is required in case of PVC resize` and we can move the discussion there. I think that issue is not part of this PR and should not be addressed here.
How does this sound to you?